server/auth/types/saml/saml_auth.ts

```diff
@@ -55,8 +55,15 @@ export class SamlAuthentication extends AuthenticationType {
 
   private redirectToLoginUri(request: KibanaRequest, toolkit: AuthToolkit) {
     const nextUrl = this.generateNextUrl(request);
+    let clearOldVersionCookie: string;
+    if (this.config.cookie.secure) {
+      clearOldVersionCookie = 'security_authentication=; Secure; HttpOnly; Path=/';
+    } else {
+      clearOldVersionCookie = 'security_authentication=; HttpOnly; Path=/';
+    }
```

Comment:
The string generation looks exactly the same. Please have a shared function.

Reply:
moved to a shared function